cmd/av/stack_reparent.go

```diff
@@ -1,24 +1,281 @@
 
+	"emperror.dev/errors"
+	"github.com/aviator-co/av/internal/actions"
+	"github.com/aviator-co/av/internal/git"
+	"github.com/aviator-co/av/internal/meta"
+	"github.com/aviator-co/av/internal/sequencer"
+	"github.com/aviator-co/av/internal/sequencer/planner"
 	"github.com/aviator-co/av/internal/utils/colors"
+	"github.com/aviator-co/av/internal/utils/stackutils"
+	"github.com/charmbracelet/bubbles/spinner"
+	tea "github.com/charmbracelet/bubbletea"
+	"github.com/charmbracelet/lipgloss"
+	"github.com/go-git/go-git/v5/plumbing"
+	"github.com/mattn/go-isatty"
 	"github.com/spf13/cobra"
 )
 
+var stackReparentFlags struct {
+	Parent   string
+	Abort    bool
+	Continue bool
+	Skip     bool
+}
+
 var stackReparentCmd = &cobra.Command{
-	Use:    "reparent <new-parent>",
-	Hidden: true,
+	Use:   "reparent",
+	Short: "Reparent branches",
 	RunE: func(cmd *cobra.Command, args []string) error {
-		_, _ = fmt.Fprint(os.Stderr,
-			colors.Failure("ERROR: "),
-			"The ", colors.CliCmd("av stack reparent"), " command is deprecated: ",
-			"use ", colors.CliCmd("av stack sync --parent <new-parent>"), " instead.",
-			"\n",
-		)
-		os.Exit(1)
+		repo, err := getRepo()
+		if err != nil {
+			return err
+		}
+
+		db, err := getDB(repo)
+		if err != nil {
+			return err
+		}
+
+		var opts []tea.ProgramOption
+		if !isatty.IsTerminal(os.Stdout.Fd()) {
+			opts = []tea.ProgramOption{
+				tea.WithInput(nil),
+			}
+		}
+		p := tea.NewProgram(stackReparentViewModel{
+			repo:    repo,
+			db:      db,
+			spinner: spinner.New(spinner.WithSpinner(spinner.Dot)),
+		}, opts...)
+		model, err := p.Run()
+		if err != nil {
+			return err
+		}
+		if err := model.(stackReparentViewModel).err; err != nil {
+			return actions.ErrExitSilently{ExitCode: 1}
+		}
+		if s := model.(stackReparentViewModel).rebaseConflictErrorHeadline; s != "" {
+			return actions.ErrExitSilently{ExitCode: 1}
+		}
 		return nil
 	},
 }
+
+type stackReparentState struct {
+	InitialBranch   string
+	NewParentBranch string
+	Seq             *sequencer.Sequencer
+}
+
+type stackReparentSeqResult struct {
+	result *git.RebaseResult
+	err    error
+}
+
+type stackReparentViewModel struct {
+	repo    *git.Repo
+	db      meta.DB
+	state   *stackReparentState
+	spinner spinner.Model
+
+	rebaseConflictErrorHeadline string
+	rebaseConflictHint          string
+	abortedBranch               plumbing.ReferenceName
+	err                         error
+}
+
+func (vm stackReparentViewModel) Init() tea.Cmd {
+	return tea.Batch(vm.spinner.Tick, vm.initCmd)
+}
+
+func (vm stackReparentViewModel) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
+	switch msg := msg.(type) {
+	case error:
+		vm.err = msg
+		return vm, tea.Quit
+	case *stackReparentState:
+		vm.state = msg
+		if stackReparentFlags.Skip || stackReparentFlags.Continue || stackReparentFlags.Abort {
+			if stackReparentFlags.Abort {
+				vm.abortedBranch = vm.state.Seq.CurrentSyncRef
+			}
+			return vm, vm.runSeqWithContinuationFlags
+		}
+		return vm, vm.runSeq
```

Comment:
skip and continue have the same behavior?

Reply:
They are different. https://git-scm.com/docs/git-rebase#_mode_options